src/app/payments/helpers.ts

```diff
@@ -0,0 +1,296 @@
+import { getTwoFALimits, getAccountLimits, MS_PER_DAY, getDealerConfig } from "@config"
+import { AccountLimitsChecker, TwoFALimitsChecker } from "@domain/accounts"
+import { LightningPaymentFlowBuilder } from "@domain/payments"
+import { ErrorLevel, ExchangeCurrencyUnit, WalletCurrency } from "@domain/shared"
+import { AlreadyPaidError } from "@domain/errors"
+import { CENTS_PER_USD } from "@domain/fiat"
+
+import { NewDealerPriceService } from "@services/dealer-price"
+import {
+  AccountsRepository,
+  WalletInvoicesRepository,
+  WalletsRepository,
+} from "@services/mongoose"
+import { LndService } from "@services/lnd"
+import { LedgerService } from "@services/ledger"
+import {
+  addAttributesToCurrentSpan,
+  asyncRunInSpan,
+  recordExceptionInCurrentSpan,
+  SemanticAttributes,
+} from "@services/tracing"
+
+import { getCurrentPrice } from "@app/prices"
+
+const usdHedgeEnabled = getDealerConfig().usd.hedgingEnabled
+
+const dealer = NewDealerPriceService()
+const ledger = LedgerService()
+
+const usdFromBtcMidPriceFn = async (
+  amount: BtcPaymentAmount,
+): Promise<UsdPaymentAmount | DealerPriceServiceError> =>
+  asyncRunInSpan(
+    "app.payments.usdFromBtcMidPriceFn",
+    {
+      [SemanticAttributes.CODE_FUNCTION]: "usdFromBtcMidPriceFn",
+      [SemanticAttributes.CODE_NAMESPACE]: "app.payments",
+    },
+    async () => {
+      const midPriceRatio = await getMidPriceRatio()
+      if (midPriceRatio instanceof Error) return midPriceRatio
+
+      const usdPaymentAmount = {
+        amount: BigInt(Math.ceil(Number(amount.amount) * midPriceRatio)),
+        currency: WalletCurrency.Usd,
+      }
+
+      addAttributesToCurrentSpan({
+        "usdFromBtcMidPriceFn.midPriceRatio": midPriceRatio,
+        "usdFromBtcMidPriceFn.incoming.amount": Number(amount.amount),
+        "usdFromBtcMidPriceFn.incoming.unit":
```

Comment:
the `incoming.unit` comparison feels off to me. 

we're looking at the `amount.currency` which is of type `BtcPaymentAmount`

something is probably off here I think

Reply:
maybe it's just because the code is duplicated below, but in that case code should be dedupe

Reply:
Not sure I understand. The goal is to get something that looks like this in honeycomb

![image](https://user-images.githubusercontent.com/17693119/164734229-661b539d-9832-4995-a589-5fc47a14cefe.png)